### ISSUE TITLE
positioning objects in the world

### ISSUE THREAD

**Issue by jw3877** [danielepanozzo/cg-old#52]
For each triangle mesh, we can:
- Upload the vertex position data to the GPU only once
- Give each instance of that object a unique model matrix to position it in the world

For small scenes, it seems okay to position the objects manually using a combination of translation, scaling, rotation matrices. But imagine a complicated scene like a bedroom where there are many objects in close proximity -- a desk with a computer monitor on it, books scattered about, objects all over the place.

Manually positioning all of these objects with (trans, scale, rotation) matrices seems like a difficult job. If something is even slightly off, it might be noticeable -- like a vase floating on top of a table.

Is there software to "drag and drop" the meshes into desired position and have it generate the initial model matrices for each object? How do they create complicated scenes like this?

**Comment by danielepanozzo** [comment]
Yes, there are many modeling software (https://en.wikipedia.org/wiki/List_of_3D_modeling_software) that can help you with scene creation. I recommend to use blender(https://www.blender.org), it is free and you can find a lot of educational material online.
